issue/9

0hun opened the pull request:
회원 수정 메소드 추가
  - 회원 수정 메소드 추가
  - BDDMockito 적용하여 테스트 코드 수정

0hun committed:
feat (feature) : 회원 수정 메소드 추가

회원 수정 메소드 추가
  - 회원 수정 메소드 추가
  - BDDMockito 적용하여 테스트 코드 수정

yhmane commented on src/main/java/com/delivery/user/controller/UserController.java:
```diff
@@ -24,48 +25,64 @@
+        if (existsUser) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).build();
+        }
+
```
이 부분은 추후에 Advice 생성후 Service 로직에 빼는게 좋을거 같아요

yhmane commented on src/test/java/com/delivery/user/controller/UserControllerTest.java:
```diff
@@ -34,172 +35,211 @@
+        // given
+        long userId = 1L;
+
+        doNothing().when(userService).delete(userId);
```
Mockitod와 BDDMockito가 혼용될 이유는 없을거 같아용
아래와 같이 대체가 가능합니다
```suggestion
willDoNothing().given(userService).delete(userId);
```

yhmane commented on src/test/java/com/delivery/user/controller/UserControllerTest.java:
```diff
@@ -34,172 +35,211 @@
+        // given
+        long userId = 1L;
+
+        doThrow(NoSuchElementException.class).when(userService).delete(userId);
```
```suggestion
       willThrow(NoSucehElementException)...
```
마찬가지로 대체가 가능하네용

yhmane commented on src/test/java/com/delivery/user/controller/UserControllerTest.java:
```diff
@@ -3,11 +3,12 @@
+import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.doThrow;
```
Mockito와 BDDMockito를 혼용해서 쓸 필요는 없어보입니다.
둘중의 하나의 api로 통일하는게 좋을거 같네요

yhmane reviewed (commented):
Mocking Test에 대한 통일이 필요해 보입니다

0hun committed:
refactor : Mockito을 BDDMockito으로 변경

Mockito을 BDDMockito으로 변경
- void method BDDMockito으로 변경
- exception 테스트 시 BDDMockito으로 변경

yhmane reviewed (approved):
Merge 하겠습니다
